audio/data.py

```diff
@@ -258,66 +262,86 @@ def __init__(self, items, path, config=AudioConfig(), **kwargs):
+
+        if self.nchannels is None: self.nchannels = item.nchannels
+            
+        if item.nchannels > 1:
+            if self.config.downmix:
+                item.sig = DownmixMono(channels_first=True)(item.sig)
+
+            elif(self.nchannels != item.nchannels):
+                raise ValueError(f'''Multiple channel sizes detected. Channel size {item.nchannels} of file 
+                                {fn} does not match others' channel size of {self.nchannels}. A dataset may
+                                not contain different number of channels. Please set downmix=true in AudioConfig or 
+                                separate files with different number of channels.''')
+
+    def add_spectro(self, item: AudioItem):
+        cache_path = self._get_cache_path(item.path)
+
+        if self.config.cache and cache_path.exists():
+            item.spectro = torch.load(cache_path)
+        else:
+            item.spectro = self.create_spectro(item)
+            if self.config.cache:
+                self._save_in_cache(item.path, item.spectro)
+
+        if self.config.duration and self.config._processed: 
+                item.spectro, item.start, item.end = tfm_crop_time(item.spectro, self.config._sr, self.config.duration, self.config.sg_cfg.hop, self.config.pad_mode)
+        return item
+
+    
+    def create_spectro(self, item: AudioItem):
+        if self.config.mfcc: 
+            mel = MFCC(sr=item.sr, n_mfcc=self.config.sg_cfg.n_mfcc, melkwargs=self.config.sg_cfg.mel_args())(item.sig)
+        else:
+            mel = MelSpectrogram(**(self.config.sg_cfg.mel_args()))(item.sig)
+            if self.config.sg_cfg.to_db_scale: 
+                mel = SpectrogramToDB(top_db=self.config.sg_cfg.top_db)(mel)
+        mel = mel.permute(0, 2, 1)
+        if self.config.standardize: 
+            mel = standardize(mel)
+        if self.config.delta: 
+            mel = torch.cat([torch.stack([m,torchdelta(m),torchdelta(m, order=2)]) for m in mel]) 
+        return mel
+
+    def _get_cache_path(self, fn):
+        folder = md5(str(asdict(self.config))+str(asdict(self.config.sg_cfg)))
+        fname = f"{md5(str(fn))}-{fn.name}.pt"
+        return Path(self.config.cache_dir/(f"{folder}/{fname}"))
+
+    def _save_in_cache(self, fn, spectro):
+        cache_path = self._get_cache_path(fn)
+        os.makedirs(cache_path.parent, exist_ok=True)
+        torch.save(spectro, cache_path)
+        _record_cache_contents(self.config, [cache_path])
 
     def get(self, i):
-        item = self.items[i]
-        if isinstance(item, AudioItem): return item
-        if isinstance(item, (str, PosixPath, Path)):
-            if not ('/') in str(item): return self.open(self.path/item)
-            else:                      return self.open(item)
-        raise TypeError(f"Can't handle type {type(item)}, only AudioItem, str, Path or PosixPath")  
+        fn = super().get(i)
+        res = self.open(fn)
```

Comment:
Why not just `return self.open(fn)`

Reply:
So you want get to look like this: 

```
    def get(self, i):
        fn = super().get(i)
        return self.open(fn)
```

Reply:
Only a small thing but saves one line :)

Reply:
Just making sure I was understanding what you were saying.  I'll change that.